core-plugins/src/main/java/io/cdap/plugin/batch/source/ExcelInputReader.java

```diff
@@ -266,23 +278,26 @@ public void transform(KeyValue<LongWritable, Object> input, Emitter<StructuredRe
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     super.configurePipeline(pipelineConfigurer);
 
-    excelInputreaderConfig.validate();
+    // Get failure collector for updated validation API
+    FailureCollector collector = pipelineConfigurer.getStageConfigurer().getFailureCollector();
+    excelInputreaderConfig.validate(collector);
 
     if (Strings.isNullOrEmpty(excelInputreaderConfig.columnList) &&
```

Comment:
this looks like it belongs in the validate() method

Reply:
Fixed